torch/_dynamo/resume_execution.py

```diff
@@ -462,15 +466,74 @@ def update(
+                        *add_push_null(
+                            [
+                                create_instruction(
+                                    "LOAD_FAST", argval="__nested_resume_fns"
+                                ),
+                                create_instruction("LOAD_CONST", argval=-1),
+                                create_instruction("BINARY_SUBSCR"),
+                            ]
+                        ),
+                        # del __nested_resume_fns[-1]
+                        create_instruction("LOAD_FAST", argval="__nested_resume_fns"),
+                        create_instruction("LOAD_CONST", argval=-1),
+                        create_instruction("DELETE_SUBSCR"),
+                        # load [__nested_resume_fns, __nested_frame_values]
+                        create_instruction("LOAD_FAST", argval="__nested_resume_fns"),
+                        create_instruction("LOAD_FAST", argval="__nested_frame_values"),
+                        create_instruction("BUILD_LIST", arg=2),
+                        # load __nested_frame_values[-1]
+                        create_instruction("LOAD_FAST", argval="__nested_frame_values"),
+                        create_instruction("LOAD_CONST", argval=-1),
+                        create_instruction("BINARY_SUBSCR"),
+                        # create [
+                        #     __nested_resume_fns,
+                        #     __nested_frame_values,
+                        #     *__nested_frame_values[-1][0],
+                        #     *__nested_frame_values[-1][1]],
+                        # ]
+                        create_dup_top(),
+                        create_instruction("LOAD_CONST", argval=0),
+                        create_instruction("BINARY_SUBSCR"),
+                        create_instruction("LIST_EXTEND", arg=2),
+                        create_instruction("LOAD_CONST", argval=1),
+                        create_instruction("BINARY_SUBSCR"),
+                        create_instruction("LIST_EXTEND", arg=1),
+                        # del __nested_frame_values[-1]
+                        create_instruction("LOAD_FAST", argval="__nested_frame_values"),
+                        create_instruction("LOAD_CONST", argval=-1),
+                        create_instruction("DELETE_SUBSCR"),
+                        # delete __nested values
+                        create_instruction("DELETE_FAST", argval="__nested_resume_fns"),
+                        create_instruction(
+                            "DELETE_FAST", argval="__nested_frame_values"
+                        ),
+                        # Set is_tracing_resume_prologue back to allow graph breaks
+                        # in the nested resume
+                        create_instruction("LOAD_CONST", argval=False),
+                        create_instruction(
+                            "STORE_FAST", argval=IS_TRACING_RESUME_PROLOGUE_VARNAME
```

Comment:
Can you just write a comment about this one? I am not sure why we need this flag.

Reply:
This is used to prevent dynamo from graph breaking in resume function prologues (defined [here](https://github.com/pytorch/pytorch/pull/159329/files#diff-957b95045ab713eecb0b16216229756cb2e314018c21db825ab01a269eeea3a1R54), set [here](https://github.com/pytorch/pytorch/pull/159329/files#diff-957b95045ab713eecb0b16216229756cb2e314018c21db825ab01a269eeea3a1R403), unset [here](https://github.com/pytorch/pytorch/pull/159329/files#diff-957b95045ab713eecb0b16216229756cb2e314018c21db825ab01a269eeea3a1R521))